src/znc.cpp

```diff
@@ -227,6 +244,17 @@ void CZNC::Loop() {
                               true);
                 }
                 break;
+            case ECONFIG_WANT_WRITE:
+                SetConfigState(ECONFIG_NOTHING);
+
+                if (GetConfigWriteDelay() > 0) {
+                  if (m_pConfigTimer == nullptr) {
+                    m_pConfigTimer = new CConfigWriteTimer(GetConfigWriteDelay());
+                    GetManager().AddCron(m_pConfigTimer);
+                  }
+                  break;
+                }
+		/* Fall through */
             case ECONFIG_NEED_WRITE:
             case ECONFIG_NEED_VERBOSE_WRITE:
```

Comment:
I think, need to stop timer here, if there was request to write config without delay. Otherwise, after the timer runs, config will be written again.